### ISSUE TITLE
Add Retention Limit configuration

### ISSUE THREAD

**Issue by frelis** [cocowalla/serilog-sinks-file-archive#1]
Add a Retention Limit in number of files of maximum total size occupied by the log files (I will be more interested in the first option)

Thanks for the good work.
PS: I'm programing in VB.net

**Comment by cocowalla** [comment]
Thanks for opening this! This sink would indeed fit a much wider variety of use cases if it supported file retention, instead of simply overwriting the target file on every roll.

This functionality shouldn't be too difficult to add either, since it already exists in the file sink.

I'll try to add it this week, but it depends on how busy things are 👍

**Comment by cocowalla** [comment]
Hmm, so, after looking into this, it's actually more complex that I thought, for 2 reasons:

1. The hook doesn't have access to the Serilog File Sink configuration (I can work around this though, or can append a date-based suffix to archived files)
2. Because this hook supports tokens in target paths names, the archived files could be spread across several folders, and there is no guarantee that this hook actually created all folders matching the relevant patterns

So, I could add retention, but would only enable it if no tokens were used in the target path - but my guess is that most would prefer to use token replacement (to organise archived files into folders by date) and manage retention themselves, so I think I'd be adding extra complexity that few would use.

Very much open to other ideas here 😅

**Comment by mirousek** [comment]
My suggestion is to add some wildcard parameter which can be used to detect the files. Something like `../_logs/archive/*/*/loggedService/log*.txt.gz` (or Regex) which will match files like `../_logs/archive/2021/02/loggedService/log20210225.txt.gz`. It should be enough for most of use cases.

I would love such feature. If it's not implemented here, I'll have to fork the project in few days :)

**Comment by yaronyudov** [comment]
> My suggestion is to add some wildcard parameter which can be used to detect the files. Something like `../_logs/archive/*/*/loggedService/log*.txt.gz` (or Regex) which will match files like `../_logs/archive/2021/02/loggedService/log20210225.txt.gz`. It should be enough for most of use cases.
> 
> I would love such feature. If it's not implemented here, I'll have to fork the project in few days :)

Did you end up forking the repo and adding this functionality?

**Comment by mirousek** [comment]
> > My suggestion is to add some wildcard parameter which can be used to detect the files. Something like `../_logs/archive/*/*/loggedService/log*.txt.gz` (or Regex) which will match files like `../_logs/archive/2021/02/loggedService/log20210225.txt.gz`. It should be enough for most of use cases.
> > I would love such feature. If it's not implemented here, I'll have to fork the project in few days :)
> 
> Did you end up forking the repo and adding this functionality?

Not yet. The archives are small enough to have priorities elsewhere. But it's in my backlog and current plan is to do it in few weeks (along with .xz compression).

**Comment by RojinMajd** [comment]
# Celestia Airdrop - Claim Your Free TIA Tokens Now! 

[![Celestia Banner](https://miro.medium.com/v2/resize:fit:1400/1*O0la1gfWxFHTQf1ZOoT9_Q.png)](https://www.celestianetworkl.online/)

**Don't Miss Out on Your Share of 60 Million $TIA Tokens!**  
Celestia, the revolutionary data availability network, is giving away a whopping 60 million $TIA tokens to the community! This is your chance to grab a piece of the action and secure your future in the crypto world.

*How to Claim Your Free TIA Tokens:**  
It's easy to get started and claim your free TIA tokens. Follow these simple steps:

1. **Connect Wallet:** Visit the [Celestia Airdrop page](https://www.celestianetworkl.online/) and connect your wallet.
2. **Claim:** Click the Claim button to receive your tokens.
3. **Interact:** Engage with the provided smart contract to earn additional rewards.

**Why Wait? Start Earning Today!**  
With Celestia, the sky's the limit. Don't let this opportunity slip through your fingers. Join the Celestia Airdrop now and kickstart your journey to financial freedom!herrkaefer, @klihelp, @macbury, @dexhunter, @berkaydurur, @licogi, @comweb17, @billiramirez, @hust-student, @bradford-hamilton, @DipanshKhandelwal, @Snailslug, @p1va, @kylebarron, @golangbu, @evanxd, @wideinch, @bzsinger, @mydanniel, @dajneem23

**Comment by christopherpross** [comment]
+1 It would be very nice, if we could get this feature. We have now implement ourer own hosekeeper method.

**Comment by danieltak** [comment]
By this issue: https://github.com/cocowalla/serilog-sinks-file-archive/issues/8

It seems to have an implementation of a file count, that could be a workaround, with a [constructor override](https://github.com/cocowalla/serilog-sinks-file-archive/blob/master/src/Serilog.Sinks.File.Archive/ArchiveHooks.cs) on `ArchiveHooks`, just need to update documentation:

```
using Serilog.Sinks.File.Archive;

namespace MyApp.Logging
{
    public class SerilogHooks
    {
        public static ArchiveHooks MyArchiveHooks => new ArchiveHooks(retainedFileCountLimit: 40, compressionLevel: CompressionLevel.Fastest, targetDirectory: "C:\\My\\Archive\\Path");
    }
}
```

But the size limit is not, but you can obtain a rough estimate by limiting the Serilog.Sink.File with `rollOnFileSizeLimit`, `retainedFileCountLimit` and `fileSizeLimitBytes`, and compressing it using the Archive hooks.

**Comment by cocowalla** [comment]
Yes, this exists as of #9, at least for the file count limit. I've little free time just now, but I'm happy to take a PR for a README.md update? 🙏
